cmd/cloud_sql_proxy/proxy.go

```diff
@@ -338,10 +338,24 @@ func CreateInstanceConfigs(dir string, useFuse bool, instances []string, instanc
 	}
 	// FUSE disabled.
 	if len(instances) == 0 && instancesSrc == "" {
+		// Failure to specifying instance can be caused by following reasons.
+		// 1. not enough information is provided by flags
+		// 2. failed to invoke gcloud
+		var flags string
 		if fuse.Supported() {
-			return nil, errors.New("must specify -projects, -fuse, or -instances")
+			flags = "-projects, -fuse, or -instances"
```

Comment:
This should be 
flags = "-projects, -fuse, -instances or -instances_metadata"
and the line below should be 
flags = "-projects, -instances or -instances_metadata"

-instances_metadata was missing earlier as well, which was wrong I think.